rotkehlchen/chain/aggregator.py

```diff
@@ -1310,18 +1311,27 @@ def check_single_address_activity(
         """Checks whether address is active in the given chains. Returns a list of active chains"""
         active_chains = []
         for chain in chains:
-            chain_manager = self.get_chain_manager(chain)
+            chain_manager: EvmManager = self.get_chain_manager(chain)
             try:
                 if chain == SupportedBlockchain.AVALANCHE:
+                    avax_manager = cast(AvalancheManager, chain_manager)
                     # just check balance and nonce in avalanche
                     has_activity = (
-                        chain_manager.w3.eth.get_transaction_count(address) != 0 or
-                        chain_manager.get_avax_balance(address) != ZERO
+                        avax_manager.w3.eth.get_transaction_count(address) != 0 or
+                        avax_manager.get_avax_balance(address) != ZERO
                     )
                     if has_activity is False:
                         continue
-                elif chain_manager.node_inquirer.etherscan.has_activity(address) is False:  # noqa: E501
-                    continue
+                else:
+                    etherscan_activity = chain_manager.node_inquirer.etherscan.has_activity(address)  # noqa: E501
+                    only_token_spam = (
+                        etherscan_activity == EtherscanHasChainActivity.TOKENS and
+                        chain_manager.transactions.address_has_been_spammed(address=address)
+                    )
+                    if only_token_spam or etherscan_activity == EtherscanHasChainActivity.NONE:
+                        continue  # do not add the address for the chain
+
+                    # else we add the chain
```

Comment:
Do not do that. It's obvious that you add the chain later. And if you **really** wanted to add this it should be in line 1339 as part of same line. Not on its own line. Please remove

```suggestion
```